src/View/Helper/PaginatorHelper.php

```diff
@@ -613,17 +614,18 @@ public function hasNext($model = null)
     /**
      * Returns true if the given result set has the page number given by $page
      *
-     * @param string|null $model Optional model name. Uses the default if none is specified.
      * @param int $page The page number - if not set defaults to 1.
+     * @param string|null $model Optional model name. Uses the default if none is specified.
      * @return bool True if the given result set has the specified page number.
      * @link https://book.cakephp.org/3.0/en/views/helpers/paginator.html#checking-the-pagination-state
+     * @throws \InvalidArgumentException
      */
-    public function hasPage($model = null, $page = 1)
+    public function hasPage($page = 1, $model = null)
```

Comment:
No type hints for the arguments?

Reply:
I didn't know how fast we now progress into the 7.1 typehints.

Reply:
We have started adding typehints to take care of #11935. So I think it would be nice to add typehints to any argument changes we are doing.

Reply:
Would the type errors give enough context for people using the old call signature?

Reply:
Yes, most likely :)

Reply:
@ADmad Feel free to add in these PRs. I just thought a more efficient way would be the sniffer run over the namespace.

Reply:
We can add typehints to the helper namespace as a separate task. This is incremental progress :+1:

Reply:
Is 7.1 still the target or more like 7.2?

Reply:
why? 7.1 is still enough right now.

Reply:
@inoas Yes 7.1 is still the target.